qiita_ware/ebi.py

```diff
@@ -543,46 +485,33 @@ def generate_run_xml(self):
             data_block = ET.SubElement(run, 'DATA_BLOCK')
             files = ET.SubElement(data_block, 'FILES')
             ET.SubElement(files, 'FILE', {
-                'filename': join(self.ebi_dir, basename(file_path)),
+                'filename': file_path,
                 'filetype': file_type,
                 'quality_scoring_system': 'phred',
                 'checksum_method': 'MD5',
                 'checksum': md5}
             )
 
-        return run_set
+        return ET.tostring(run_set)
 
-    def generate_submission_xml(self, action):
+    def generate_submission_xml(self, date_to_hold=date.today()):
```

Comment:
IMHO, the `date_to_hold` parameter should be renamed. To me, this name implies that the release should be held until today.

Also, this should default to `None`, then in the body of the function it should be set to `date.today()` (or today + 365 days) if it is `None`. Otherwise, this will not behave as you expect it to. As it is now, `date.today()` will be called only once, when the `def` is first executed (so, only once each time Qiita is launched, I assume), and the `date_to_hold` will be the same for all submissions, regardless of the actual submission date.